multiwallet.go

```diff
@@ -0,0 +1,428 @@
+			return nil, errors.E("settings db is in use by another process")
+		}
+		return nil, errors.E("error opening settings db store: %s", err.Error())
+	}
+
+	db, err := storm.Open(filepath.Join(rootDir, walletsDbName))
+	if err != nil {
+		log.Errorf("Error opening wallet database: %s", err.Error())
+		if err == bolt.ErrTimeout {
+			// timeout error occurs if storm fails to acquire a lock on the database file
+			return nil, errors.E("wallet database is in use by another process")
+		}
+		return nil, errors.E("error opening wallet index database: %s", err.Error())
+	}
+
+	// init database for saving/reading wallet objects
+	err = db.Init(&Wallet{})
+	if err != nil {
+		log.Errorf("Error initializing wallet database: %s", err.Error())
+		return nil, err
+	}
+
+	syncData := &syncData{
+		syncCanceled:          make(chan bool),
+		syncProgressListeners: make(map[string]SyncProgressListener),
+	}
+
+	mw := &MultiWallet{
+		dbDriver:  dbDriver,
+		rootDir:   rootDir,
+		db:        db,
+		configDB:  configDB,
+		activeNet: activeNet,
+		wallets:   make(map[int]*LibWallet),
+		syncData:  syncData,
+	}
+
+	mw.listenForShutdown()
+
+	loadedWallets, err := mw.loadWallets()
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("Loaded %d wallets", loadedWallets)
+
+	return mw, nil
+}
+
+func (mw *MultiWallet) Shutdown() {
+	log.Info("Shutting down dcrlibwallet")
+
+	// Trigger shuttingDown signal to cancel all contexts created with `contextWithShutdownCancel`.
+	mw.shuttingDown <- true
+
+	mw.CancelSync()
+
+	for _, w := range mw.wallets {
+		w.Shutdown()
+	}
+
+	if logRotator != nil {
+		log.Info("Shutting down log rotator")
+		logRotator.Close()
+	}
+
+	if mw.db != nil {
+		err := mw.db.Close()
+		if err != nil {
+			log.Errorf("db closed with error: %v", err)
+		} else {
+			log.Info("db closed successfully")
+		}
+	}
+}
+
+func (mw *MultiWallet) loadWallets() (int, error) {
+	query := mw.db.Select(q.True()).OrderBy("ID")
+	var wallets []Wallet
+	err := query.Find(&wallets)
+	if err != nil && err != storm.ErrNotFound {
+		return 0, err
+	}
+
+	mw.wallets = make(map[int]*LibWallet)
+	for _, w := range wallets {
+		w.walletDbDriver = mw.dbDriver
+		w.netType = mw.activeNet.Name
+		libWallet, err := NewLibWallet(&w)
+		if err != nil {
+			return 0, err
+		}
+
+		mw.wallets[w.ID] = libWallet
+	}
+
+	return len(wallets), nil
+}
+
+func (mw *MultiWallet) GetBackupsNeeded() int32 {
+	var backupsNeeded int32
+	for _, w := range mw.wallets {
+		if w.WalletOpened() && w.wallet.Seed != "" {
+			backupsNeeded++
+		}
+	}
+
+	return backupsNeeded
+}
+
+func (mw *MultiWallet) LoadedWalletsCount() int32 {
+	return int32(len(mw.wallets))
+}
+
+func (mw *MultiWallet) OpenedWalletsRaw() []int {
+	wallets := make([]int, 0)
+	for _, w := range mw.wallets {
+		if w.WalletOpened() {
+			wallets = append(wallets, w.wallet.ID)
+		}
+	}
+
+	return wallets
+}
+
+func (mw *MultiWallet) OpenedWallets() string {
```

Comment:
`OpenedWalletIDs`
Is it not possible to access the other method with GoMobile? Would modifying the return type of that method to `[]int32` help?